### PR TITLE
HTTPS all the things

### PR DESCRIPTION
**Pull request by claudijd**
Noticed that we still have a bunch of HTTP links, which could present mild MiTM risks that we can totally remove.

One short coming of this was:

cargo/src/etc/_cargo:13 "http://zsh.sourceforge.net/Doc/Release/Completion-System.html#Completion-Functions" of which does not have an HTTPS equivalent yet.

### CODE REVIEW

File: src/cargo/ops/common_for_install_and_uninstall.rs [hunk 1]
```diff
@@ -92,7 +92,7 @@ where
                                 "the `--vers` provided, `{}`, is \
                                        not a valid semver version requirement\n\n
                                        Please have a look at \
-                                       http://doc.crates.io/specifying-dependencies.html \
+                                       https://doc.crates.io/specifying-dependencies.html \
```
Comment (pietroalbini) [inline src/cargo/ops/common_for_install_and_uninstall.rs]:
```suggestion
                                       https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html \
```

While you're making the changes can you also point this to the canonical URL?

File: src/cargo/ops/cargo_package.rs [hunk 2]
```diff
@@ -152,7 +152,7 @@ fn check_metadata(pkg: &Package, config: &Config) -> CargoResult<()> {
 
         config.shell().warn(&format!(
             "manifest has no {things}.\n\
-             See http://doc.crates.io/manifest.html#package-metadata for more info.",
+             See https://doc.rust-lang.org/cargo/reference/manifest.html for more info.",
```
Comment (dwijnand) [inline src/cargo/ops/cargo_package.rs]:
```suggestion
             See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info.",
```

File: src/cargo/ops/registry.rs [hunk 3]
```diff
@@ -290,7 +290,7 @@ fn transmit(
                      the following are not valid badges and were ignored: {}. \
                      Either the badge type specified is unknown or a required \
                      attribute is missing. Please see \
-                     http://doc.crates.io/manifest.html#package-metadata \
+                     https://doc.rust-lang.org/cargo/reference/manifest.html \
```
Comment (dwijnand) [inline src/cargo/ops/registry.rs]:
```suggestion
                     https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata \
```

File: src/bin/cargo/commands/pkgid.rs [hunk 4]
```diff
@@ -27,7 +27,7 @@ Example Package IDs
      crates.io/foo                | foo    | *         | *://crates.io/foo
      crates.io/foo#1.2.3          | foo    | 1.2.3     | *://crates.io/foo
      crates.io/bar#foo:1.2.3      | foo    | 1.2.3     | *://crates.io/bar
-     http://crates.io/foo#1.2.3   | foo    | 1.2.3     | http://crates.io/foo
+     https://crates.io/foo#1.2.3  | foo    | 1.2.3    | https://crates.io/foo
```
Comment (dwijnand) [inline src/bin/cargo/commands/pkgid.rs]:
```suggestion
     https://crates.io/foo#1.2.3  | foo    | 1.2.3     | https://crates.io/foo
```

**Comment by dwijnand** [comment]
I can follow-up.

**Comment by claudijd** [comment]
@dwijnand fixed

File: tests/testsuite/build_auth.rs [hunk 5]
```diff
@@ -105,7 +105,7 @@ fn http_auth_offered() {
             authors = []
 
             [dependencies.bar]
-            git = "http://127.0.0.1:{}/foo/bar"
+            git = "https://127.0.0.1:{}/foo/bar"
```
Comment (dwijnand) [inline tests/testsuite/build_auth.rs]: This causes the error to be "SSL error: received early EOF; class=Net (12); code=Eof (-20)"

Comment (claudijd) [reply tests/testsuite/build_auth.rs]: Fixed in ee20f73

File: tests/testsuite/build_auth.rs [hunk 6]
```diff
@@ -126,11 +126,11 @@ fn http_auth_offered() {
         .with_status(101)
         .with_stderr_contains(&format!(
             "\
-[UPDATING] git repository `http://{addr}/foo/bar`
```
Comment (ehuss) [inline tests/testsuite/build_auth.rs]: I don't think this test should be changed.

File: tests/testsuite/build_auth.rs [hunk 7]
```diff
@@ -171,7 +171,7 @@ fn https_something_happens() {
             authors = []
 
             [dependencies.bar]
-            git = "https://127.0.0.1:{}/foo/bar"
+            git = "http://127.0.0.1:{}/foo/bar"
```
Comment (ehuss) [inline tests/testsuite/build_auth.rs]: This test shouldn't be modified, either.

**Comment by ehuss** [comment]
Thanks!

@bors r+

**Pull request merged**